app/models/competition.rb

```diff
@@ -739,11 +741,11 @@ def trainee_delegate_ids
   end
 
   def enable_v2_registrations!
-    update_column :uses_v2_registrations, true
+    update_column :registration_version, 'v2'
   end
 
   def uses_new_registration_service?
-    self.uses_v2_registrations
+    self.registration_version == 'v2'
```

Comment:
See my other comment above: If you use `prefix` on the enum, you can change this to `self.registration_version_v2?`